test/schema/ListPublicEvents.schema.js

```diff
@@ -0,0 +1,35 @@
+const eventsSchema = {
```

Comment:
the schema is incomplete, you need to add the two properties at the root level of the response: status and body. Also the payload property is an object, it means that you need to specify the fields of that object in the schema.

Reply:
The **response.data** returns an array of jsons, and each one has a payload propertie which can variate depending of the event. So, how i do it in that case?

Reply:
well the schema defines all the possible fields, but it does not mean that all the fields in the schema should be present in the response, for that, there is a property called required that specifies which fields are always required

Reply:
The thing is the payload properties always changes depending of the type event. There is no properties in common. For example, for the push event, the payload has this properties: push_id, size, distinct_size, ref, head, before and commits; for the release event, the payload has this properties: action and release. As you can see, these two events have no payload properties in common. So, in order to solve this, could i generate diferents schemas and use one depending of a property value (in this case, the type event)?

Reply:
choose one of those types. as I said before, if you specify the property but is not in the required list, the test will not fail

Reply:
I ran the test and throws me this 
![image](https://user-images.githubusercontent.com/46982083/177434324-391a9302-ea98-4409-a5a8-e44accb0b0f2.png)

Reply:
so i searched and found this in a stackoverflow forum: https://stackoverflow.com/questions/62495169/axios-post-error-about-circular-structure
![image](https://user-images.githubusercontent.com/46982083/177434420-ba27bd72-47eb-450f-a280-f334f750d261.png)
so what should i do to read the whole response? Reading the response properties one by one? Like: response.status, response.headers and so on...

Reply:
Is not that hard, not sure why you are getting that, so I will give you a couple of hints:

1. In file test/GithubApi.Contract.test.js, you should not do anything distinct from the provided in the guide. That means that line 22, should be `expect(response).to.be.jsonSchema(listPublicEventsSchema)`
2. The current eventsSchema that is displayed in this PR, should be changed to include two new properties at the root level (body and status). And in the payload, you can specify the fields for one of the event types, whichever you want. So general structure, that you should complete is something similar to this:
    ```
    const eventsSchema = {
      type: 'object',
      properties: {
        status: {
          type: 'number'
        },
        body: {
          type: 'array',
          items: {
    ....here the structure you currently have + more detail in payload field....
          }
        },
        required: ['status', 'body']
      }
    };
    ```

Reply:
It finally works. Thanks for your help <3 @holgiosalos